### PR TITLE
feat: [PLATO-409] Segment.io integration

### PR DESCRIPTION
**Pull request by mattvanvoorst-contentful**
**_What will change?_**

**Segment/typewriter tracking**
Added Segment / Typewriter for analytics purposes

More intricate documentation is available through the readme file that was added in the analytics folder, but in a nutshell:
- Added a boilerplate for generating a typewriter.yml file, which is then used to generate a typed typewriter.analytics object.
- Sentry is set uo to report any Segment schema violation
- The configuration of Segment is driven by environment variables
- The typewriter schema is automatically regenerated on various events

**Consent management**
- Added @contentful/experience-consent-manager for Contentful's Osano based consent management
- Built a custom RQ hook that creates a ConsentManager instance and exposes it globally
- Added a little component that injects the custom styling, and adds a button to open the consent settings drawer

Ticket: https://contentful.atlassian.net/browse/PLATO-409
Docs: https://contentful.atlassian.net/wiki/spaces/PROD/pages/4138500311/Templates+-+Segment+tracking

**_Learnings_**
* Typewriter and Segment are reliant on each other, but diverge a lot in types and package versions
* Typewriter's config doesn't really accommodate for different workspaces in Segment. 

**_Testing steps_**
* Follow the instructions to set up the additional configurations in the `analytics/readme` file
* Please check out the branch, run `yarn install` and confirm if the process completes successfully.
* Confirm the `trackingPlans` property has a valid ID in the generates `typewriter.yml` file
* Confirm `yarn typewriter:update` runs successfully
* (_optional_) Attempt to send a typewriter event to Segment.

### CODE REVIEW

**Comment by stephanLeece** [comment]
Codewise it's lookin good! i'll test it locally now

**Comment by stephanLeece** [comment]
Should `SEGMENT_TRACKINGPLAN_ID` be added to .env.example?

**Comment by stephanLeece** [comment]
Test resultz:

- [x] set up the additional configurations in the analytics/readme file
- [x] check out the branch, run yarn install and confirm if the process completes successfully.
```
Typewriter config exists -- continuing to update
Attempting to update tracking plan
Successfully created typewriter config
```

- [x] Confirm the trackingPlans property has a valid ID in the generates typewriter.yml file
I see a ` typewriter.yml` file in my repo root, with a trackingPlans.id property.

- [x] Confirm yarn typewriter:update runs successfully
```
Loading tracking plans... Loaded
Generating files... Done
Running After Script: ./node_modules/.bin/prettier --write ./analytics/plan.json... Done
```

**Review by stephanLeece — APPROVED** [review]
Nice!

**Comment by mattvanvoorst-contentful** [comment]
> 

Decided against it for now! Because we didn't add them for the Sentry stuff either

File: package.json [hunk 1]
```diff
@@ -53,11 +56,13 @@
     "@graphql-codegen/cli": "2.13.12",
     "@graphql-codegen/client-preset": "1.1.4",
     "@graphql-codegen/introspection": "2.2.1",
+    "@oclif/core": "^2.0.7",
```
Comment (AdrianLThomas) [inline package.json:59]: As discussed on Slack, to be removed :-)

File: analytics/readme.MD [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+Where to find the values is documented on https://contentful.atlassian.net/wiki/spaces/PROD/pages/edit-v2/4138500311.
+
+### Related files
+
+The aforementioned variables are used in a few places:
+
+- `src/_ctf-private/ctf-analytics` contains the logic that inject the tracking snippet and initialises the Typewriter
+  config. It also sets up error handling
+- `analytics` contains all files generates by Typewriter, this readme and some utility file
+- `segment.example.yml` the template file that is going to generate our `segment.yml` file
```
Comment (AdrianLThomas) [inline analytics/readme.MD:54]: I think you mean `typewriter.example.yml`?

Comment (AdrianLThomas) [reply analytics/readme.MD:54]: and typewriter.yml

**Comment by AdrianLThomas** [comment]
Looks like you're missing the update to `.env.example` too

**Pull request merged**